liferay-appsec/liferay-portal
# LPD-6871 Allow Site Administrators to create Forgot Password Utility Page

[LPD-6871](https://liferay.atlassian.net/browse/LPD-6871)

Allow Site Administrators to create Forgot PasswordUtility Page

### Changes
The UP has been created using the Utility Page framework.
Changes have been made so that the UP is related to `UpdatePasswordAction` (action is in the core of the portal, not in a module). This way, when using the "Forgot Password" functionality, it checks if Forgot Password UP exists in order to show it or not. 

### Test

- Create a Forgtot Password Utility Page (UP)
- Edit the UP and add the sign in widget.
- When the UP is created, is not set as default. 
- I follow the steps to recover the password and receive the mail
- I access the url indicated in the mail. The following cases occur:
  - If there is no UP, the current behavior of the portal is shown without the changes.
  - If the UP exists but it is not assigned by default, 
  - If UP exists and is assigned by default, the UP is displayed.
